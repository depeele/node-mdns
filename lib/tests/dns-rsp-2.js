var Util    = require('util'),
    Utils   = require('../utils.js'),
    Message = require('../message.js'),

    data    = [
        // Header
        0xf3, 0x49, 0x81, 0x80, 0x00, 0x01, 0x00, 0x01,
        0x00, 0x02, 0x00, 0x00,
        
        // Question
        0x04, 0x70, 0x6f, 0x70, 0x64, 0x02, 0x69, 0x78,
        0x06, 0x6e, 0x65, 0x74, 0x63, 0x6f, 0x6d, 0x03,
        0x63, 0x6f, 0x6d, 0x00, 0x00, 0xff, 0x00, 0x01,

        // Answer
        0xc0, 0x0c, 0x00, 0x0f, 0x00, 0x01, 0x00, 0x00,
        0x03, 0x10, 0x00, 0x18, 0x00, 0x0a, 0x06, 0x6e,
        0x6f, 0x6d, 0x61, 0x69, 0x6c, 0x09, 0x65, 0x61,
        0x72, 0x74, 0x68, 0x6c, 0x69, 0x6e, 0x6b, 0x03,
        0x6e, 0x65, 0x74, 0x00, 0xc0, 0x11, 0x00, 0x02,
        0x00, 0x01, 0x00, 0x00, 0x03, 0x10, 0x00, 0x0b,
        0x08, 0x73, 0x63, 0x72, 0x61, 0x74, 0x63, 0x68,
        0x79, 0xc0, 0x39, 0xc0, 0x11, 0x00, 0x02, 0x00,
        0x01, 0x00, 0x00, 0x03, 0x10, 0x00, 0x08, 0x05,
        0x69, 0x74, 0x63, 0x68, 0x79, 0xc0, 0x39  
 
    ];

/*****************************************************************************
 * Unpack test
 *
 */
console.log("--------------------------------------------------------------");
var rBuf    = new Buffer( data ),
    message = new Message( {unpack:rBuf} );

console.log("Unpack mesage from %d bytes of raw data:", message.consumed);
console.log("message:\n%s", Util.inspect(message, false, 20));

//process.exit(0);

/*****************************************************************************
 * Pack test
 *
 */
console.log("--------------------------------------------------------------");
var wBuf    = new Buffer( 4096 );

message.pack(wBuf);

console.log("Packed message (%d bytes):\n%s",
            message.produced, Utils.buf2hex(wBuf, {
                                            length:     message.produced,
                                            octetsPer:  16
                             }));

if (message.produced !== data.length)
{
    console.log("*** Pack error: length %d != expected %d",
                message.produced, data.length);
    //process.exit(-1);
}

var diff    = 0;
for (var idex = 0, len = data.length; idex < len; idex++)
{
    if (wBuf[idex] !== data[idex])
    {
        diff++;
        console.log("*** Pack error: @ %s: %s != expected %s",
                    Utils.int2hex(idex, 2),
                    Utils.int2hex(wBuf[idex]),
                    Utils.int2hex(data[idex]));

        if (diff > 8)   { break; }
    }
}

if (diff === 0)
{
    console.log(">>> packed buffer matches original source");
}
else
{
    if (diff > 8)   {console.log("*** ..."); }

    console.log("*** packed buffer mismatch.");
    console.log("Original source (%d bytes):\n%s",
                rBuf.length, Utils.buf2hex(rBuf, {
                                            length:     rbuf.length,
                                            octetsPer:  16
                             }));
}
