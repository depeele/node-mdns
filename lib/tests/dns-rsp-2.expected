--------------------------------------------------------------
Unpack mesage from 115 bytes of raw data:
message:
{ header: 
   { id: 62281,
     qr: 1,
     opcode: 0,
     aa: 0,
     tc: 0,
     rd: 1,
     ra: 1,
     z: 0,
     ad: 0,
     cd: 0,
     rcode: 0,
     qdCount: 1,
     anCount: 1,
     nsCount: 2,
     arCount: 0,
     consumed: 12 },
  question: 
   [ { qname: 'popd.ix.netcom.com.',
       qtype: 255,
       qclass: 1,
       consumed: 24 } ],
  answer: 
   [ { name: 'popd.ix.netcom.com.',
       type: 15,
       class: 1,
       ttl: 784,
       rdata: 
        { preference: 10,
          exchange: 'nomail.earthlink.net.',
          consumed: 26 },
       consumed: 36 } ],
  authority: 
   [ { name: 'ix.netcom.com.',
       type: 2,
       class: 1,
       ttl: 784,
       rdata: { ns: 'scratchy.earthlink.net.', consumed: 13 },
       consumed: 23 },
     { name: 'ix.netcom.com.',
       type: 2,
       class: 1,
       ttl: 784,
       rdata: { ns: 'itchy.earthlink.net.', consumed: 10 },
       consumed: 20 } ],
  additional: [],
  consumed: 115 }
--------------------------------------------------------------
Header: flags[ 0x8180 ]: qr[ 1 ], opcode[ 0 ], aa[ 0 ], tc[ 0 ], rd[ 1 ], ra[ 1 ], z[ 0 ], ad[ 0 ], cd[ 0 ], rcode[ 0 ]
Packed message (115 bytes):
 0x0000: f3 49 81 80 00 01 00 01 00 02 00 00 04 70 6f 70 
 0x0010: 64 02 69 78 06 6e 65 74 63 6f 6d 03 63 6f 6d 00 
 0x0020: 00 ff 00 01 c0 0c 00 0f 00 01 00 00 03 10 00 18 
 0x0030: 00 0a 06 6e 6f 6d 61 69 6c 09 65 61 72 74 68 6c 
 0x0040: 69 6e 6b 03 6e 65 74 00 c0 11 00 02 00 01 00 00 
 0x0050: 03 10 00 0b 08 73 63 72 61 74 63 68 79 c0 39 c0 
 0x0060: 11 00 02 00 01 00 00 03 10 00 08 05 69 74 63 68 
 0x0070: 79 c0 39
>>> packed buffer matches original source
